src/test/projects/javacc-generate/pom.xml

```diff
@@ -0,0 +1,66 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>test.javacc.spoon</groupId>
+  <artifactId>foo</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <properties>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+  <dependencies>
+  </dependencies>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>javacc-maven-plugin</artifactId>
+        <version>2.6</version>
+        <executions>
+          <execution>
+            <id>javacc</id>
+            <goals>
+              <goal>javacc</goal>
+            </goals>
+          </execution>
+        </executions>
+        <dependencies>
+          <dependency>
+            <groupId>net.java.dev.javacc</groupId>
+            <artifactId>javacc</artifactId>
+            <version>7.0.6</version>
+          </dependency>
+        </dependencies>
+      </plugin>
+    <plugin>
+      <groupId>fr.inria.gforge.spoon</groupId>
+      <artifactId>spoon-maven-plugin</artifactId>
+      <version>3.5-SNAPSHOT</version>
```

Comment:
Why don't you use a filtered resource for example?

See:
* https://github.com/xwiki/xwiki-commons/blob/46e0ef7112577803a00e0c7f1abc035be38a8bf2/xwiki-commons-tools/xwiki-commons-tool-xar/xwiki-commons-tool-xar-integration-tests/pom.xml#L55
* https://github.com/xwiki/xwiki-commons/blob/0a56fb5fb8c59ea93bc09a5897c929322b64265a/xwiki-commons-tools/xwiki-commons-tool-xar/xwiki-commons-tool-xar-integration-tests/src/test/resources/wrongVersion/pom.xml

Reply:
because i normally use gradle and didn't knew it. Going to have a look thx.

Reply:
What about simply removing ` <version>3.5-SNAPSHOT</version>` as you did above?

Reply:
If you remove the version the build won't be reproducible and maven will use the latest version available. You need a version (and you should actually use the maven enforcer to always make sure all versions are set).